Releng 879/build.yml

99 opened the pull request:
Final "green" pipeline is here https://github.com/hashicorp/consul-terraform-sync/actions/runs/1170051340

Enabling CRT build.yml to CTS.

This includes files and logic to build all binaries, docker images and rpms/debs and attach to the GitHub workflow.

We should have 21 artifacts in total:

consul-terraform-sync-0.3.0.arm.rpm
consul-terraform-sync-0.3.0.arm64.rpm
consul-terraform-sync-0.3.0.x86_64.rpm

consul-terraform-sync_0.3.0_amd64.deb
consul-terraform-sync_0.3.0_arm.deb
consul-terraform-sync_0.3.0_arm64.deb

consul-terraform-sync_0.3.0_docker_linux_386.tar
consul-terraform-sync_0.3.0_docker_linux_amd64.tar
consul-terraform-sync_0.3.0_docker_linux_arm.tar 
consul-terraform-sync_0.3.0_docker_linux_arm64.tar

consul-terraform-sync_0.3.0_freebsd_386.zip
consul-terraform-sync_0.3.0_linux_386.zip
consul-terraform-sync_0.3.0_windows_386.zip
consul-terraform-sync_0.3.0_darwin_amd64.zip
consul-terraform-sync_0.3.0_freebsd_amd64.zip
consul-terraform-sync_0.3.0_linux_amd64.zip
consul-terraform-sync_0.3.0_linux_arm64.zip
consul-terraform-sync_0.3.0_linux_armelv5.zip
consul-terraform-sync_0.3.0_linux_armhfv6.zip
consul-terraform-sync_0.3.0_solaris_amd64.zip
consul-terraform-sync_0.3.0_windows_amd64.zip

Things outstanding (will be covered in a separate PR):

- Create a ci.hcl to enable the promotion pipeline (https://github.com/hashicorp/consul-terraform-sync/pull/363).
- Backport these changes to any active release branches and ensure that the version in version.go correctly reflects the version that branch is targeting.

Commit:
Add build.yml

Commit:
Update build.yml

Commit:
test

Commit:
test

Commit:
test

Commit:
test

Commit:
Test build.yml with more changes

Commit:
Test build.yml with more changes

Commit:
Test build.yml with more changes

Commit:
Test build.yml with more changes

Commit:
Test build.yml with more changes

Commit:
Test build.yml with more changes

Commit:
Test build.yml with more changes

Commit:
Test build.yml with more changes

Commit:
Test build.yml with more changes

Commit:
test generate_metadata

Commit:
test generate_metadata

Commit:
add scripts folder

Commit:
add scripts folder

Commit:
test

Commit:
Add upload-artifact action

Commit:
Add upload-artifact action

claire-labry committed:
adding docker and docker step

claire-labry committed:
adding docker step in build.yml

claire-labry committed:
testing

claire-labry committed:
moving docker files into the actual docker directory lol

claire-labry committed:
removing extra underscore

claire-labry committed:
removing sneaky underscore

claire-labry committed:
getting into dectective mode

claire-labry committed:
testing

Commit:
test version changes

Commit:
test version changes

Commit:
comment workflow

Commit:
fix version

Commit:
comment workflow

Commit:
comment workflow

Commit:
fixups

Commit:
Add arm part

Commit:
Add arm part

Commit:
Add arm part

Commit:
Add arm now

Commit:
Add darwin

claire-labry committed:
adding docker steps for amd64 and build-arm

claire-labry committed:
modifying the Dockerfile to match CTS

claire-labry committed:
adding fixes to Dockerfile and adding docker-entrypoint.sh

Commit:
test zip staff

Commit:
test

Commit:
Add ci.hcl

samsalisbury committed:
add linux packaging skeleton

samsalisbury committed:
add package github action

Commit:
cleanup

Commit:
update ci.hcl

Commit:
update ci.hcl

Commit:
update ci.hcl

hashicorp-cla commented:
[![CLA assistant check](https://cla.hashicorp.com/pull/badge/not_signed)](https://cla.hashicorp.com/hashicorp/consul-terraform-sync?pullRequest=362)


Thank you for your submission! We require that all contributors sign our Contributor License Agreement ("CLA") before we can accept the contribution. [Read and sign the agreement](https://cla.hashicorp.com/hashicorp/consul-terraform-sync?pullRequest=362)

[Learn more about why HashiCorp requires a CLA and what the CLA includes](https://www.hashicorp.com/cla)


---
**3** out of **4** committers have signed the CLA.
- [x] claire-labry
- [x] sarahethompson
- [x] samsalisbury
- [ ] Natalie Serebryakova

----

**Natalie Serebryakova** seems not to be a GitHub user.
You need a GitHub account to be able to sign the CLA.
If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).


<sub>Have you signed the CLA already but the status is still pending? [Recheck it](https://cla.hashicorp.com/check/hashicorp/consul-terraform-sync?pullRequest=362).</sub>

Commit:
re-trigger gh checks

Commit:
Fix yml syntax for Package

sarahethompson committed:
Adding docker buildx to build the multi arch images properly
Some updates to build.yml to make sure we're uplaoding all artifacts (rpms + debs included)
Build arm5 and arm6

Commit:
Update the branch name

Commit:
Delete ci.hcl

sarahethompson committed:
Upping version number to 0.3.0 on main.
This should maybe be 0.3.1 going forward and then on the release/0.3.x branch we bump to 0.3.0 - will check with team.

sarahethompson committed:
fixing typo

sarahethompson committed:
Adding the branch to the metadata.json as the orchestrator needs that now.

samsalisbury committed:
linux packages: update homepage and license

sarahethompson committed:
fix arm package names

claire-labry reviewed (approved):
🎉

sarahethompson committed:
updating arch for arm docker image to linux/arm/v6

findkim reviewed (approved):
The files look good. Thanks for all the work around onboarding this project! Looking forward to learning more offline.

sarahethompson committed:
fixing arm/v6 docker build/save
